### PR TITLE
update sample url for prep of rename of repo url

### PR DESCRIPTION
**Pull request by nabobalis**
So we can get ready for upstream changes aka https://github.com/sunpy/sample-data/issues/25

### CODE REVIEW

**Review by ConorMacBride — COMMENTED** [review]
This looks good. Maybe a changelog to alert people that the endpoint is going to change in case they are using the URL elsewhere?

**Comment by nabobalis** [comment]
> This looks good. Maybe a changelog to alert people that the endpoint is going to change in case they are using the URL elsewhere?

Yes fair point.

File: changelog/6289.trivial.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+The sample-data URL will be changing from ``https://github.com/sunpy/sample-data/raw/master/sunpy/v1/`` to ``https://github.com/sunpy/data/raw/main/sunpy/v1/``.
+GitHub will keep the redirect alive for sometime but will eventually expire it.
```
Comment (Cadair) [inline changelog/6289.trivial.rst:2]:
```suggestion
The primary sample-data URL will be changing from ``https://github.com/sunpy/sample-data/raw/master/sunpy/v1/`` to ``https://github.com/sunpy/data/raw/main/sunpy/v1/``.
GitHub will keep the redirect alive for sometime but will eventually expire it.
The ``data.sunpy.org`` mirror will continue to be available.
```

**Review by Cadair — APPROVED** [review]
I tested this locally, it doesn't cause any obvious errors with the new URL not working, it just slows the download down a little.

File: changelog/6289.trivial.rst [hunk 2]
```diff
@@ -0,0 +1,3 @@
+The primary sample-data URL will be changing from ``https://github.com/sunpy/sample-data/raw/master/sunpy/v1/`` to ``https://github.com/sunpy/data/raw/main/sunpy/v1/``.
+GitHub will keep the redirect alive for sometime but will eventually expire it.
```
Comment (Cadair) [inline changelog/6289.trivial.rst:2]:
```suggestion
We expect GitHub to redirect from the old URL for sometime but will eventually expire it.
```

**Pull request merged**